heatmap-client/src/canvas/geometry.rs

```diff
@@ -25,45 +25,96 @@ const COLOR_VERTICES: &[Vertex] = &[
 
 const COLOR_INDICES: &[u16] = &[0, 2, 3, 0, 2, 1];
 pub struct Geometry {
-    pub blend_vertex_buffer: wgpu::Buffer,
-    pub blend_index_buffer: wgpu::Buffer,
-    pub blend_num_indices: u32,
-    pub color_vertex_buffer: wgpu::Buffer,
-    pub color_index_buffer: wgpu::Buffer,
-    pub color_num_indices: u32,
+    pub lod_0_layer: BufferLayer,
+    pub lod_1_layer: BufferLayer,
+    pub lod_2_layer: BufferLayer,
+    pub color_layer: BufferLayer,
+}
+
+pub struct BufferLayer {
+    pub vertex_buffer: wgpu::Buffer,
+    pub index_buffer: wgpu::Buffer,
+    pub num_indices: u32,
 }
 
 impl Geometry {
-    pub fn generate_buffers(render_context: &RenderContext, buffer_data: BufferStorage) -> Self {
+    pub fn generate_buffers(
+        render_context: &RenderContext,
+        buffer_data: Vec<BufferStorage>,
+    ) -> Self {
         //////////////////////////////
         // Set up buffers to render //
         //////////////////////////////
 
-        let blend_vertex_buffer =
+        // LOD 0
+        let lod_0_vertex_buffer =
```

Comment:
It might make sense to store these all in a vec, allowing for a variable amount of LOD levels

Reply:
and it would reduce code duplication

Reply:
This is done!

Reply:
you can generate the lod vertex buffers themselves using a map statement to further reduce code duplication if you want